public/app/features/query/state/queryAnalytics.ts

```diff
@@ -65,17 +63,17 @@ export function emitDataRequestEvent(datasource: DataSourceApi) {
     const totalQueries = Object.keys(queryCacheStatus).length;
     const cachedQueries = Object.values(queryCacheStatus).filter((val) => val === true).length;
 
-    eventData.panelId = data.request!.panelId;
-    eventData.dashboardId = data.request!.dashboardId;
+    eventData.panelId = data.request?.panelId;
     eventData.totalQueries = totalQueries;
     eventData.cachedQueries = cachedQueries;
 
     const dashboard = getDashboardSrv().getCurrent();
     if (dashboard) {
-      eventData.dashboardId = dashboard.id;
```

Comment:
Enterprise analytics is using this but it should be fine, should just treat as 0
https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/analytics/eventmodels/models.go

question is if this will break any analytics dashboards.

dam this does not look good. 
https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/analytics/summaries/database/database.go

Reply:
I put back all this stuff -- we can look at analytics later :(